src/controller/java/src/chip/onboardingpayload/QRCodeOnboardingPayloadParser.kt

```diff
@@ -118,5 +85,38 @@ class QRCodeOnboardingPayloadParser(private val mBase38Representation: String) {
       index.addAndGet(numberOfBitsToRead)
       return dest
     }
+
+    fun extractPayload(inString: String): String {
+      var chipSegment = ""
+      val delimiter = '%'
+      val startIndices = mutableListOf<Int>()
+      startIndices.add(0)
+
+      for (i in inString.indices) {
```

Comment:
I wonder if kotlin can be made shorter. maybe using `forEachIndexed` or even:

```kotlin
startIndices.addAll(
  inString
        .mapIndexed { i, c -> Pair(i,c)}
        .filter { v -> v.second == delimiter }
        .map { v -> v.first + 1 }
)
```

overall also the logic seems to be 'split by delimiters' so why are we not doing:

```kotlin
for (segment in inString.split('%')) {
  ...
}
```

Reply:
After this I tried to make the function cleaner, see above. Overall I believe kotlin code should not use indices/string operations as much and that may make things cleaner.